agent-control/src/agent_control/config_validator/k8s.rs

```diff
@@ -0,0 +1,146 @@
+use crate::agent_control::config::AgentControlDynamicConfig;
+
+use super::{DynamicConfigValidator, DynamicConfigValidatorError};
+
+pub struct K8sReleaseNamesConfigValidator<V: DynamicConfigValidator> {
+    inner: V,
+    ac_release_name: String,
+    cd_release_name: String,
+}
+
+impl<V: DynamicConfigValidator> DynamicConfigValidator for K8sReleaseNamesConfigValidator<V> {
+    fn validate(
+        &self,
+        dynamic_config: &AgentControlDynamicConfig,
+    ) -> Result<(), DynamicConfigValidatorError> {
+        self.inner.validate(dynamic_config)?;
+
+        dynamic_config
+            .agents
+            .keys()
+            .try_for_each(|agent_id| match agent_id.as_str() {
+                agent_id if agent_id == self.ac_release_name => {
+                    Err(validation_error(agent_id, "agent-control-deployment"))
```

Comment:
Shouldn't we be using the actual `self.ac_release_name` instead of `"agent-control-deployment"` and same for CD?